### PR TITLE
Shoatman client capabilities

### PR DESCRIPTION
**Pull request by shoatman**
Added support for client capabilities to config
Corrected behavior relative to claims parameters and acquireTokenSilent
Added tests for operation parameters relative to adding claims to acquireTokenSilent Calls

### CODE REVIEW

File: msal/src/main/java/com/microsoft/identity/client/claims/ClaimsRequest.java [hunk 1]
```diff
@@ -46,6 +46,28 @@
     private List<RequestedClaim> mAccessTokenClaimsRequested = new ArrayList<>();
     private List<RequestedClaim> mIdTokenClaimsRequested = new ArrayList<>();
 
+    @Override
```
Comment (kreedula) [inline msal/src/main/java/com/microsoft/identity/client/claims/ClaimsRequest.java]: nit : `CheckStyle` and `PMD` will fail for these generated methods. Add `@SuppressWarnings("PMD")` and 

`//CHECKSTYLE:OFF` above the method and `//CHECKSTYLE:ON` at the end of the method

File: msal/src/main/java/com/microsoft/identity/client/claims/ClaimsRequest.java [hunk 2]
```diff
@@ -46,6 +46,28 @@
...
+
+        if (mUserInfoClaimsRequested != null ? !mUserInfoClaimsRequested.equals(that.mUserInfoClaimsRequested) : that.mUserInfoClaimsRequested != null)
+            return false;
+        if (mAccessTokenClaimsRequested != null ? !mAccessTokenClaimsRequested.equals(that.mAccessTokenClaimsRequested) : that.mAccessTokenClaimsRequested != null)
+            return false;
+        return mIdTokenClaimsRequested != null ? mIdTokenClaimsRequested.equals(that.mIdTokenClaimsRequested) : that.mIdTokenClaimsRequested == null;
+    }
+
+    @Override
+    public int hashCode() {
```
Comment (kreedula) [inline msal/src/main/java/com/microsoft/identity/client/claims/ClaimsRequest.java]: Same here w.r.t to `PMD` and `Checkstyle`

File: msal/src/main/java/com/microsoft/identity/client/claims/RequestedClaim.java [hunk 3]
```diff
@@ -41,6 +41,24 @@ public String getName() {
         return mName;
     }
 
+    @Override
+    public boolean equals(Object o) {
```
Comment (kreedula) [inline msal/src/main/java/com/microsoft/identity/client/claims/RequestedClaim.java]: Same here w.r.t to `PMD` and `Checkstyle`

File: msal/src/main/java/com/microsoft/identity/client/claims/RequestedClaimAdditionalInformation.java [hunk 4]
```diff
@@ -46,6 +46,27 @@
...
+        return mValue != null ? mValue.equals(that.mValue) : that.mValue == null;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = mEssential != null ? mEssential.hashCode() : 0;
+        result = 31 * result + (mValues != null ? mValues.hashCode() : 0);
+        result = 31 * result + (mValue != null ? mValue.hashCode() : 0);
+        return result;
+    }
```
Comment (kreedula) [inline msal/src/main/java/com/microsoft/identity/client/claims/RequestedClaimAdditionalInformation.java]: Same here w.r.t to `PMD` and `Checkstyle`

File: msal/src/test/res/raw/aad_capabilities_cp1.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
+{
+  "client_id" : "4b0db8c2-9f26-4417-8bde-3f0e3656f8e0",
+  "authorization_user_agent" : "DEFAULT",
+  "redirect_uri" : "msauth://com.microsoft.identity.client.sample.local/1wIqXSqBj7w%2Bh11ZifsnqwgyKrY%3D",
+  "multiple_clouds_supported":true,
+  "broker_redirect_uri_registered": true,
+  "account_mode": "MULTIPLE",
+  "client_capabilities": "CP1",
```
Comment (kreedula) [inline msal/src/test/res/raw/aad_capabilities_cp1.json]: nit: remove `comma` we have only one capability.

File: testapps/testapp/src/main/res/raw/msal_config.json [hunk 6]
```diff
@@ -3,6 +3,7 @@
   "authorization_user_agent" : "DEFAULT",
   "redirect_uri" : "msauth://com.microsoft.identity.client.sample.local/1wIqXSqBj7w%2Bh11ZifsnqwgyKrY%3D",
   "multiple_clouds_supported":true,
+  "client_capabilities": "CP1",
```
Comment (kreedula) [inline testapps/testapp/src/main/res/raw/msal_config.json]: same here

File: msal/src/main/java/com/microsoft/identity/client/internal/controllers/OperationParametersAdapter.java [hunk 7]
```diff
@@ -202,6 +223,23 @@ public static AcquireTokenOperationParameters createAcquireTokenOperationParamet
         return acquireTokenOperationParameters;
     }
 
+    public static ClaimsRequest addClientCapabilitiesToClaimsRequest(ClaimsRequest cr, String clientCapabilities){
+
+        ClaimsRequest mergedClaimsRequest = null;
```
Comment (kreedula) [inline msal/src/main/java/com/microsoft/identity/client/internal/controllers/OperationParametersAdapter.java]: nit : instantiation and assignment can be combined

`final ClaimsRequest mergedClaimsRequest = (cr == null) ? new ClaimsRequest() : cr;`

File: msal/src/test/res/raw/aad_capabilities_cp1.json (hunk 5 above)
Comment (shoatman) [reply msal/src/test/res/raw/aad_capabilities_cp1.json]: Comma is needed to separate client_capabilities from authorities.  No comma in the value for client_capabilities.

File: testapps/testapp/src/main/res/raw/msal_config.json (hunk 6 above)
Comment (shoatman) [reply testapps/testapp/src/main/res/raw/msal_config.json]: Same as above

File: msal/src/main/java/com/microsoft/identity/client/internal/controllers/OperationParametersAdapter.java (hunk 7 above)
Comment (shoatman) [reply msal/src/main/java/com/microsoft/identity/client/internal/controllers/OperationParametersAdapter.java]: Done.

File: msal/src/main/java/com/microsoft/identity/client/claims/RequestedClaimAdditionalInformation.java (hunk 4 above)
Comment (shoatman) [reply msal/src/main/java/com/microsoft/identity/client/claims/RequestedClaimAdditionalInformation.java]: Done

Comment (shoatman) [reply msal/src/main/java/com/microsoft/identity/client/claims/RequestedClaimAdditionalInformation.java]: Done

File: msal/src/main/java/com/microsoft/identity/client/internal/controllers/OperationParametersAdapter.java (hunk 7 above)
Comment (shoatman) [reply msal/src/main/java/com/microsoft/identity/client/internal/controllers/OperationParametersAdapter.java]: Done

**Pull request merged**